Jb dk welcome screen

dkimlim opened the pull request:
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

This PR improves the UI of the homepage screen when a user first opens the app. There is one clear button to "Create New List" or a more discreet link to "join an existing list". This link will redirect to a new page to enter a token that is not the same as yours stored in local storage. 

## Related Issue

closes issue #15 

## Acceptance Criteria

- [x] The welcome screen (the screen a user sees before they’ve created their list) looks something like the following (or equivalent, use your creativity!)
![68095346-759abe80-fe5d-11e9-8c9f-22879fbad048](https://user-images.githubusercontent.com/33106448/80063646-59223980-8504-11ea-81c6-2682fbde3876.png)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
<img width="718" alt="Screen Shot 2020-04-23 at 1 48 00 AM" src="https://user-images.githubusercontent.com/33106448/80063709-7f47d980-8504-11ea-9854-09c901ada017.png">

### After
Home page:
<img width="721" alt="Screen Shot 2020-04-23 at 1 34 57 AM" src="https://user-images.githubusercontent.com/33106448/80063664-5fb0b100-8504-11ea-8c75-caac906599c8.png">

Join List page (desktop + flexin' mobile 😎):
<img width="1440" alt="Screen Shot 2020-04-24 at 9 34 57 PM" src="https://user-images.githubusercontent.com/33106448/80268355-63fbdc00-8674-11ea-9522-001c513699cf.png">
<img width="505" alt="Screen Shot 2020-04-24 at 9 35 11 PM" src="https://user-images.githubusercontent.com/33106448/80268363-68c09000-8674-11ea-985f-75a2fbca5b1a.png">

jonbascos committed:
Removed /FirebaseTesting from the return.  Added Welcome heading to the top and moved the Create Shopping List button below the instructions to "Create a shopping list" to get started.

dkimlim committed:
Only committing estimates.js

dkimlim committed:
Merge branch 'master' of https://github.com/the-collab-lab/tcl-5-smart-shopping-list into dk-estimates

Updated estimates.js

jonbascos committed:
Moved Home page into the Switch to prevent Home from always being displayed.

jonbascos committed:
Changed history.push(/Home) to history.push("/").

jonbascos committed:
Imported JoinList page.  Created JoinList Route and added handleChange, checkToken and inputToken props to JoinList.  Moved handleChange and checkToken from Home and into App so JoinList could pull data from App. Also imported Firebase.

jonbascos committed:
Created JoinList to keep things seperate from Home. Still need to work on passing Props again.  May need to have Lauren help us with that unless Diane can help.

jonbascos committed:
Cleaned up the CSS of things we were not using anymore.  Kept App.css and made everything centered by using Flexbox.

jonbascos committed:
Cbanged .navbutton to make is a little smaller and with rounded corners.

dkimlim committed:
Merge branch 'jb-dk-welcome-screen' of https://github.com/the-collab-lab/tcl-5-smart-shopping-list into jb-dk-welcome-screen

dkimlim committed:
Update JoinList to pass token

jonbascos committed:
added a className to the button element

jonbascos committed:
moved logic from the top to App.js

jonbascos committed:
added Join-List-Button styling

jonbascos committed:
Modified checkToken to redirect to ShoppingList.  Imported useHistory.

jonbascos committed:
Added Home class for styling

jonbascos committed:
Cleanup of unused items

dkimlim committed:
Modified css of homepage

dkimlim committed:
Remove footer if token is not available like on homepage

dkimlim committed:
Modify CSS to add fixed Header throughout the App, and modify Home container

jonbascos committed:
Added styling for Join List button.  Widened the Enter Token Input field.  Added heading and background.

jonbascos committed:
created joinList-Container and JoinList-text classes to style their respective items.

laurenmbeatty commented:
This is so gorgeous.  I love it. The 'Create a New List' button animation is so nice.

laurenmbeatty commented on src/App.css:
```diff
@@ -1,107 +1,26 @@
+    width: 100vw;
+    padding:0;
+    background-color: white;
+    max-height: 100px;
```
If you get rid of the max-height, it will still look the same, and respond better with the window re-sizing (just allows the text to wrap around).

dkimlim replied:
got it!

laurenmbeatty commented on src/components/HeaderLogo.js:
```diff
@@ -0,0 +1,12 @@
+
+const Header = () => {
+    return (
+        <div className="Header">
```
You could/should just make this a `<header>` tag instead of a `div`.

dkimlim replied:
true that! good catch on semantics. Must. Refrain. Using. Divs.

laurenmbeatty commented on src/components/HeaderLogo.js:
```diff
@@ -0,0 +1,12 @@
+const Header = () => {
+    return (
+        <div className="Header">
+            <h1>Smart Shopping List</h1>
```
`h1` perfect!

dkimlim committed:
Normalize css names and add a bit of style updates

Used CamelCase for naming elements and added
specificity with -element. ie. Home-container.

dkimlim committed:
Update changes with Lauren's first comments

Added header semantic in HeaderLogo.js component, and removed
max-height in header's css

dkimlim commented on src/css/Home.css:
```diff
@@ -0,0 +1,75 @@
+.Home,
+.JoinList-container {
```
We ended up including the CSS of the joinlist page because we realized it was pulling from Home.css even though it wasn't imported. 

Actually, all of the .css files were being imported throughout the app, so not sure if we really need a specific .css folder or just one massive App.css??

AshlingH replied:
Might be good to have one central CSS file for sure!

dkimlim commented on src/css/Home.css:
```diff
@@ -0,0 +1,75 @@
+}
+
+h1.Home-h1 {
+    margin-top: 0;
```
example, the h1 was pulling the margin-top from AddItemForm.css, so I had to add individual classes

laurenmbeatty commented:
I'm seeing an issue if you're on a particularly long shopping list. 
![image](https://user-images.githubusercontent.com/10378331/80269098-5c3b3800-8672-11ea-9b08-f691029d7928.png)
I'll leave some quick comments on a straightforward fix, but you may want to finesse it. I think it works fine with a small list, as well.

laurenmbeatty commented on src/App.css:
```diff
@@ -1,107 +1,25 @@
-  text-align: center;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
```
```suggestion
    justify-content: space-between;
```
This will help make sure modal shows.

dkimlim replied:
Good catch!

laurenmbeatty commented on src/App.css:
```diff
@@ -1,107 +1,25 @@
+}
+
+.Header {
+    position: fixed;
```
```suggestion
    position: sticky;
```
This will also help make sure the modal and filter/search shows.

AshlingH replied:
I have never heard of sticky!

dkimlim replied:
Yep this definitely helped!

AshlingH reviewed (approved):
This is really impressive, well done. I am in agreement with having one css file.

jendevelops reviewed (approved):
Looking good! The app is getting a lot more streamlined now!

joeylaguna commented:
This looks so great 😄 🎉

dkimlim committed:
Fixing Header to sticky

dkimlim reviewed (commented):
Made the changes for the Header so that the shopping list Search modal will show in a long list!

laurenmbeatty reviewed (approved):
Great!  Now I can see the delete item and the list filter!
